install/kubernetes/cilium/templates/cilium-configmap.yaml

```
@@ -835,8 +835,8 @@ data:
   enable-k8s-event-handover: "true"
 {{- end }}
 
-{{- if hasKey .Values "crdWaitTimeout" }}
-  crd-wait-timeout: {{ .Values.crdWaitTimeout | quote }}
+{{- if .Values.crdWaitTimeout }}
+  crd-wait-timeout: {{ include "validateDuration" .Values.crdWaitTimeout | quote }}
 {{- else if ( ne $crdWaitTimeout "5m" ) }}
   crd-wait-timeout: {{ $crdWaitTimeout | quote }}
 {{- end }}
```

After re-reading, when `.Values.crdWaitTimeout` evaluate to `false` (e.g. empty string), then the second condition will always evaluate to true (i.e. `( ne $crdWaitTimeout "5m" )`) and we will set the `crd-wait-timeout` option.

I think we should get rid of the second branch
```suggestion
{{- if .Values.crdWaitTimeout }}
  crd-wait-timeout: {{ include "validateDuration" .Values.crdWaitTimeout | quote }}
{{- end }}
```

---

good point!